DESIGN-PROPOSALS.md

```diff
@@ -0,0 +1,93 @@
+
+The purpose of the design proposal processes is to reduce the amount of "siloed knowledge" in a community. By moving decisions from a smattering of mailing lists, video calls, slack messages, Github exchanges, and hallway conversations into a well tracked artifact, the process aims to enhance communication and discoverability.
+
+## (Optional) User/User Story
+
+Detail the things that people will be able to do with the feature if it is implemented. Include as much detail as possible so that people can understand the "how" of the system. This can also be combined with the background/motivation above.
+
+## Goals
+
+The desired goal and list goals that the design achieves. What is it trying to achieve? How will you know that it has succeeded?
+
+## Non-Goals
+
+Describe what is out of scope for the design proposal. Listing non-goals helps to focus discussion and make progress.
+
+## Proposal
+
+This is where we get down to the specifics of what the proposal actually is. It should have enough detail that reviewers can understand exactly what you're proposing, but should not include things like API designs or implementation. Describe the desired outcome and how to measure success.
+
+## Design Details
+
+This section should contain enough information that the specifics of the change are understandable. This may include API specs (though not always required) or even code snippets. If there's any ambiguity about HOW your proposal will be implemented, this is the place to discuss them. This can also be combined with the proposal section above. It should also address how the solution is backward compatible and how to deal with these incompatibilities, possibly with defaulting or migrations.
+
+## Impacts / Key Questions
+
+List crucial impacts and key questions, some of which may still be open. They likely require discussion and are required to understand the trade-offs of the design. During the lifecycle of a design proposal, discussion on design aspects can be moved into this section. After reading through this section, it should be possible to understand any potentially negative or controversial impact of the design. It should also be possible to derive the key design questions: X vs Y.
+
+This will also help people understand the caveats to the proposal, other important details that didn't come across above, and alternatives that could be considered. It can also be a good place to talk about core concepts and how they relate. It can be helpful to explicitly list the pros and cons of each decision
+
+### Pros
+
+### Cons
+
+## Risks and Mitigations
+
+Describe the risks of this proposal and how they can be mitigated. This should be broadly scoped and describe how it will impact the larger ecosystem. It should include drawbacks to the proposed solution and consider the security implications of the changes too.
```

Comment:
```suggestion
Describe the risks of this proposal and how they can be mitigated. This should be broadly scoped and describe how it will impact the larger ecosystem and potentially adopters of the project; such as if adopters need to immediately update, or support a new port or protocol. It should include drawbacks to the proposed solution and consider the security implications of the changes too. 

When attempting to identify security implications of the changes, consider the following questions:
* Does the change alter the permissions or access of users, services, components - this could be an improvement or downgrade or even just a different way of doing it?
* Does the change alter the flow of information, events, and logs stored, processed, or transmitted?
* Does the change increase the 'surface area' exposed - meaning, if an operator of the project or user were to go rogue or be uninformed in its operation, do they have more areas that could be manipulated unfavorably?
* What existing security features, controls, or boundaries would be affected by this change? ```

Reply:
let's create a new section for security